connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLUtils.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.ml
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.connect.proto
+import org.apache.spark.ml.param.{ParamMap, Params}
+import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.connect.planner.SparkConnectPlanner
+import org.apache.spark.sql.connect.service.SessionHolder
+
+object MLUtils {
+
+  def setInstanceParams(instance: Params, paramsProto: proto.Params): Unit = {
+    import scala.collection.JavaConverters._
+    paramsProto.getParamsMap.asScala.foreach { case (paramName, paramValueProto) =>
+      val paramDef = instance.getParam(paramName)
+      val paramValue = parseParamValue(paramDef.paramValueClassTag.runtimeClass, paramValueProto)
+      instance.set(paramDef, paramValue)
```

Comment:
what if we make `instance.set` accept different values types here?

```
(paramDef, paramValue) match {
   case (p: DoubleParam, v: Float) => instance.set(p, v.toDouble)
   case (p: DoubleParam, v: Double) => instance.set(p, v)
   case (p: LongParam, v: Short) => instance.set(p, v.toLong)
   case (p: LongParam, v: Int) => instance.set(p, v.toLong)
   case (p: LongParam, v: Long) => instance.set(p, v)
   ...
   case _ => instance.set(paramDef, paramValue)
}
```

Reply:
This way we cannot support param that defined as Param[XXX] class.

Reply:
yes, you are right. The 3-rd lib may directly use Param[Int]